### PR TITLE
Updated to allow nonEvent fields to be evaluation only

### PR DESCRIPTION
**Pull request by ivakegg**

### CODE REVIEW

**Review by apmoriarty — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexIndexExpansionVisitor.java [hunk 1]
```diff
@@ -607,8 +607,11 @@ public boolean isExpandable(ASTERNode node) throws TableNotFoundException, JavaR
     public boolean mustExpand(ASTERNode node) throws TableNotFoundException {
         String fieldName = JexlASTHelper.getIdentifier(node);
 
-        // if the identifier is a non-event field, then we must expand it
-        return helper.getNonEventFields(config.getDatatypeFilter()).contains(fieldName);
+        // If the identifier is an index-only field, then we must expand it.
+        // This used to check for nonEvent fields, but we decided that tokenization
+        // can be handled not at evaluation time, so we only need to force index-only
```
Comment (hlgp) [inline warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexIndexExpansionVisitor.java]: not -> now?

File: warehouse/query-core/src/test/java/datawave/query/RegexQueryTest.java [hunk 2]
```diff
@@ -86,13 +97,35 @@ public void testReverse() throws Exception {
...
+        String regex = "'.*?o'";
+        for (final TestCities city : TestCities.values()) {
+            String query = CityField.CITY.name() + " == " + "'" + city.name() + "'" + AND_OP + CityField.ACCESS.name() + RE_OP + regex;
+            runTest(query, query);
+        }
+    }
+
+    @Test(expected = DatawaveFatalQueryException.class)
+    public void testIndexOnlyFails() throws Exception {
+        log.info("------  testNonEvent  ------");
```
Comment (hlgp) [inline warehouse/query-core/src/test/java/datawave/query/RegexQueryTest.java]: testIndexOnly?

File: warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexIndexExpansionVisitor.java (hunk 1 above)
Comment (ivakegg) [reply warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RegexIndexExpansionVisitor.java]: done

File: warehouse/query-core/src/test/java/datawave/query/RegexQueryTest.java (hunk 2 above)
Comment (ivakegg) [reply warehouse/query-core/src/test/java/datawave/query/RegexQueryTest.java]: done

**Review by apmoriarty — APPROVED** [review]

**Comment by ivakegg** [comment]
Already merged in separately by @hlgp

**Pull request closed without merging**